### DISCUSSION TITLE
Error: Failed to obtain access token when | when login in with OIDC provider keycloak

### DISCUSSION THREAD

**Discussion by Grafnus** [outline/outline#4452] [Self Hosting] [answered]
**To Reproduce**
You would need a keycloak instance. 
As far as I am aware, the issue lies in on of the dependencies (basically the dependency of a dependency). Therefor I will cut this step short:

1. Deploying Outline following a tutorial. Everythin work. Portainer gives thumbs up. I can acces Minio etc.
2. Adding my Keycloak instance as OIDC provider
3. starting the docker compose stack
4. trying to log into outline.
5. Getting the foling error in the web ui: `Authentication failed – we were unable to sign you in at this time. Please try again.`
6. Docker logs put out this error:
  ```
  {
  "error":"Failed to obtain access token",
  "stack":"InternalOAuthError: Failed to obtain access token
      at OAuth2Strategy._createOAuthError (/opt/outline/node_modules/passport-oauth2/lib/strategy.js:423:17)
      at /opt/outline/node_modules/passport-oauth2/lib/strategy.js:177:45
      at /opt/outline/node_modules/oauth/lib/oauth2.js:191:18
      at ClientRequest.<anonymous> (/opt/outline/node_modules/oauth/lib/oauth2.js:162:5)
      at ClientRequest.emit (node:events:526:28)
      at ClientRequest.emit (node:domain:475:12)
      at TLSSocket.socketErrorListener (node:_http_client:442:9)
      at TLSSocket.emit (node:events:526:28)
      at TLSSocket.emit (node:domain:475:12)
      at emitErrorNT (node:internal/streams/destroy:157:8)
      at emitErrorCloseNT (node:internal/streams/destroy:122:3)
      at processTicksAndRejections (node:internal/process/task_queues:83:21)",
  "level":"error",
  "message":"Error during authentication"
  }
  ```

**Expected behavior**

Seeing the default outline overview

**Screenshots**
If applicable, add screenshots or videos to help explain your problem.

None would be helpful I think

**Outline (please complete the following information):**
- Install: self hosted
- Version: 31d5119b4296366c70a2d377cb9caa59f287cc7cd88f00127e9141228d53e4d6

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser Opera
 - Version newest

**Related Threads

This is a general thread with a workaround that could be added to the dockerfile while waiting for the fixes to come throuh:
[stackoverflow](https://stackoverflow.com/questions/21129989/internaloautherror-failed-to-obtain-access-token)

This is the issue in the direct dependency of outline. I checked the version. They are using the old version of [node-oauth](https://github.com/ciaranj/node-oauth):
[passport-oauth2](https://github.com/jaredhanson/passport-oauth2/issues/163)

**Closure

I am not sure if it is even an option to change some sourcecode in a node module. But it is the only thing I could find that helps.

**Comment by tommoor** [comment, marked as answer]
Moved to discussion as this is not a bug with Outline, it's a problem with your configuration. I would recommend checking the logs of Keycloak, it should say why the connection is being refused. If there are no logs in keycloak then Outline cannot access it, it's likely a problem with the docker networking setup.

In the past this has been DNS – https://github.com/outline/outline/discussions/3501

**Reply by Grafnus** [reply]
Ok,
you where right. For everyone seeing this, I'll link you the discussion thread that held the information I needed [here](https://github.com/outline/outline/discussions/4144#discussioncomment-3770146).

Thanks for the push in the right direction.

Grafnus
